journal/src/main/scala/com/evolutiongaming/kafka/journal/Journals.scala

```diff
@@ -107,6 +157,41 @@ object Journals {
       conversionMetrics = conversionMetrics)
   }
 
+  def make[
+    F[_]
+    : Async
+    : FromTry : Fail : LogOf
+    : KafkaConsumerOf : KafkaProducerOf : HeadCacheOf : RandomIdOf
+    : MeasureDuration
+    : JsonCodec
+  ](
+     origin: Option[Origin],
+     producer: Producer[F],
+     consumer: Resource[F, Consumer[F]],
+     eventualJournal: EventualJournal[F],
+     headCache: HeadCache[F],
+     consumerPoolSize: ConsumerPoolSize,
+     consumerPoolMetrics: Option[RecoveryConsumerPoolMetrics[F]],
+     log: Log[F],
+     conversionMetrics: Option[ConversionMetrics[F]]
+  ): Resource[F, Journals[F]] = {
+    implicit val fromAttempt: FromAttempt[F]   = FromAttempt.lift[F]
+    implicit val fromJsResult: FromJsResult[F] = FromJsResult.lift[F]
+
+    for {
+      poolSize    <- consumerPoolSize.calculate.toResource
+      _           <- log.debug(s"Creating a consumer pool of size $poolSize").toResource
+      pool        <- ResourcePool.fixedSize[F, Consumer[F]](consumer, poolSize)
```

Comment:
not sure you safely can use `consumer` here, you might need to add more functionality to `release`, please check whether you need to `reset` consumer's state?

Reply:
Thanks for the comment, will check